x-pack/platform/plugins/shared/onechat/public/application/components/tools/table/tools_table_search.tsx

```diff
@@ -0,0 +1,138 @@
+import { useOnechatTools } from '../../../hooks/tools/use_tools';
+
+const getToolsTableSearchConfig = ({
+  includeSystemTools,
+  matchesByType,
+  matchesByTag,
+  tags,
+}: {
+  includeSystemTools: boolean;
+  matchesByType: Record<ToolType, number>;
+  matchesByTag: Record<string, number>;
+  tags: string[];
+}): EuiSearchBarProps => ({
+  box: {
+    incremental: true,
+    placeholder: labels.tools.searchToolsPlaceholder,
+  },
+  filters: [
+    {
+      type: 'field_value_selection',
+      field: 'type',
+      name: labels.tools.typeFilter,
+      multiSelect: false,
+      options: [
+        {
+          value: ToolType.esql,
+          name: labels.tools.esqlLabel,
+          view: (
+            <ToolFilterOption
+              name={labels.tools.esqlLabel}
+              matches={matchesByType[ToolType.esql] ?? 0}
+            />
+          ),
+        },
+        ...(includeSystemTools
+          ? [
+              {
+                value: ToolType.builtin,
+                name: labels.tools.builtinLabel,
+                view: (
+                  <ToolFilterOption
+                    name={labels.tools.builtinLabel}
+                    matches={matchesByType[ToolType.builtin] ?? 0}
+                  />
+                ),
+              },
+            ]
+          : []),
+      ],
+    },
+    {
+      type: 'field_value_selection',
+      field: 'tags',
+      name: labels.tools.tagsFilter,
+      multiSelect: 'or',
+      options: tags.map((tag) => ({
+        value: tag,
+        name: tag,
+        view: <ToolFilterOption name={tag} matches={matchesByTag[tag] ?? 0} />,
+      })),
+      searchThreshold: 1,
+    },
+  ],
+});
+
+export interface ToolsTableSearch {
+  searchConfig: Search;
+  results: ToolDefinitionWithSchema[];
+}
+
+export const useToolsTableSearch = (): ToolsTableSearch => {
+  const { addErrorToast } = useToasts();
+  const { includeSystemTools } = useToolsPreferences();
+  const { tools } = useOnechatTools({ includeSystemTools });
+  const { tags } = useToolTags({ includeSystemTools });
+  const [results, setResults] = useState<ToolDefinitionWithSchema[]>(tools);
+
+  useEffect(() => {
+    setResults(tools);
+  }, [tools]);
+
+  const handleChange = useCallback(
+    ({ query, error: searchError }: EuiSearchBarOnChangeArgs) => {
+      if (searchError) {
+        addErrorToast({
+          title: labels.tools.searchToolsErrorToast,
+          text: searchError.message,
+        });
+        return;
+      }
+
+      const newItems = query
+        ? EuiSearchBar.Query.execute(query, tools, {
+            defaultFields: ['id', 'description', 'type'],
+          })
+        : tools;
+
+      setResults(newItems);
+    },
+    [tools, addErrorToast]
+  );
+
+  const matchesByType = useMemo(() => {
+    return countBy(tools, 'type') as Record<ToolType, number>;
+  }, [tools]);
+
+  const matchesByTag = useMemo(() => {
+    return countBy(tools, 'tags');
```

Comment:
sth wrong here, tags is an array will countby still work for multivalued fields?

see screenshot I feel like it should be 3 and 1 but it's showing 2 and 0

<img width="1392" height="580" alt="Image" src="https://github.com/user-attachments/assets/aed5f790-947f-44bf-a172-0306957cf107" />

Reply:
Right, for some reason I thought it'd flatten the list but it just stores the collection as a key. Will fix by doing a `flatMap` on the tags before counting:

```suggestion
    return countBy(tools.flatMap((tool) => tool.tags));
```